### PR TITLE
Refactor ExecutionWorker

### PR DESCRIPTION
**Pull request by rzadp**
A little more clarity in preparation for multiple executors.

### CODE REVIEW

File: universal-login-relayer/lib/core/services/messages/ExecutionWorker.ts [hunk 1]
```diff
@@ -28,18 +29,28 @@ class ExecutionWorker {
...
-    do {
-      const nextMessage = await this.executionQueue.getNext();
-      if (nextMessage && this.messageExecutor.canExecute(nextMessage)){
+    const tick = async () => {
+      if (this.state === 'stopping'){
+        this.state = 'stopped';
+      } else {
+        await sleep(this.tick);
+      }
+    };
```
Comment (marekkirejczyk) [inline universal-login-relayer/lib/core/services/messages/ExecutionWorker.ts]: Should we move/extract function to class?

File: universal-login-relayer/lib/core/services/messages/ExecutionWorker.ts [hunk 2]
```diff
@@ -28,18 +29,28 @@ class ExecutionWorker {
...
+
+    const tryExecute = async (nextMessage: QueueItem) => {
+      if (this.messageExecutor.canExecute(nextMessage)){
         await this.execute(nextMessage.hash);
-      } else if (nextMessage) {
+      } else {
         await this.executionQueue.remove(nextMessage.hash);
+      }
+    };
```
Comment (marekkirejczyk) [inline universal-login-relayer/lib/core/services/messages/ExecutionWorker.ts]: Should we move/extract function to class?

File: universal-login-relayer/lib/core/services/messages/ExecutionWorker.ts [hunk 3]
```diff
@@ -28,18 +29,28 @@ class ExecutionWorker {
...
+
+    const tryExecute = async (nextMessage: QueueItem) => {
+      if (this.messageExecutor.canExecute(nextMessage)){
         await this.execute(nextMessage.hash);
-      } else if (nextMessage) {
+      } else {
         await this.executionQueue.remove(nextMessage.hash);
+      }
```
Comment (marekkirejczyk) [inline universal-login-relayer/lib/core/services/messages/ExecutionWorker.ts]: btw should we also store error if we can't execute?

**Comment by marekkirejczyk** [comment]
LGTM, 3 comments to be potentatially addressed in following PR

File: universal-login-relayer/lib/core/services/messages/ExecutionWorker.ts (hunk 1 above)
Comment (rzadp) [reply universal-login-relayer/lib/core/services/messages/ExecutionWorker.ts]: Addressed by https://github.com/UniversalLogin/UniversalLoginSDK/pull/1278

File: universal-login-relayer/lib/core/services/messages/ExecutionWorker.ts (hunk 2 above)
Comment (rzadp) [reply universal-login-relayer/lib/core/services/messages/ExecutionWorker.ts]: Addressed by https://github.com/UniversalLogin/UniversalLoginSDK/pull/1278

**Pull request merged**